addons/website_links/static/src/js/website_links_charts.js

```diff
@@ -1,238 +1,197 @@
-                });
-        };
-
-        var last_month_clicks_by_country = function () {
-            var interval = moment().subtract(30, 'days').format("YYYY-MM-DD");
-            return rpc.query({
-                    model: 'link.tracker.click',
-                    method: 'read_group',
-                    args: [[links_domain, ['create_date', '>', interval]], ['country_id']],
-                    kwargs: {groupby: 'country_id'},
-                    context: weContext.get(), // TODO use this._rpc
-                });
-        };
-
-        $.when(total_clicks(),
-               clicks_by_day(),
-               clicks_by_country(),
-               last_week_clicks_by_country(),
-               last_month_clicks_by_country())
-        .done(function (total_clicks, clicks_by_day, clicks_by_country, last_week_clicks_by_country, last_month_clicks_by_country) {
+        this.links_domain = ['link_id', '=', parseInt(link_id)];
 
-            if (total_clicks) {
+        $.when(this._totalClicks(),
+               this._clicksByDay(),
+               this._clicksByCountry(),
+               this._lastWeekClicksByCountry(),
+               this._lastMonthClicksByCountry())
+        .then(function (_totalClicks, _clicksByDay, _clicksByCountry, _lastWeekClicksByCountry, _lastMonthClicksByCountry) {
+            if (_totalClicks) {
+                this.charts = {};
                 var formatted_clicks_by_day = {};
                 var begin_date, end_date;
-                for (var i = 0 ; i < clicks_by_day.length ; i++) {
-                    var date = moment(clicks_by_day[i]['create_date:day'], "DD MMMM YYYY");
+                for (var i = 0 ; i < _clicksByDay.length ; i++) {
+                    var date = moment(_clicksByDay[i]['create_date:day'], 'DD MMMM YYYY');
                     if (i === 0) { begin_date = date; }
-                    if (i === clicks_by_day.length - 1) { end_date = date; }
-                    formatted_clicks_by_day[date.format("YYYY-MM-DD")] = clicks_by_day[i]['create_date_count'];
+                    if (i === _clicksByDay.length - 1) { end_date = date; }
+                    formatted_clicks_by_day[date.format('YYYY-MM-DD')] = _clicksByDay[i]['create_date_count'];
                 }
 
                 // Process all time line chart data
                 var now = moment();
-
-                charts.all_time_bar = new BarChart('#all_time_clicks_chart', begin_date, now, formatted_clicks_by_day);
+                this.charts.all_time_bar = new BarChart('#all_time_clicks_chart', begin_date, now, formatted_clicks_by_day);
 
                 // Process month line chart data
                 begin_date = moment().subtract(30, 'days');
-                charts.last_month_bar = new BarChart('#last_month_clicks_chart', begin_date, now, formatted_clicks_by_day);
+                this.charts.last_month_bar = new BarChart('#last_month_clicks_chart', begin_date, now, formatted_clicks_by_day);
```

Comment:
@kea14 On your next tasks, please be careful with `this`. Here you cannot use it, this is a reference to the anonymous function you are in. Here, what you wanted to do is define `var self = this;` outside of the function then use `self` in it.